grails-app/migrations/0.9.x/changelog-2023-11-06-1330-add-not-nullable-constraint-on-name-column-for-role-table.xml

```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog/1.9"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog/1.9 http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-1.9.xsd">
+
+
+  <changeSet author="drodzewicz" id="061120231330-0">
+    <preConditions onFail="MARK_RAN" onError="CONTINUE">
+      <sqlCheck expectedResult="YES">
+        SELECT IS_NULLABLE
+        FROM INFORMATION_SCHEMA.COLUMNS
```

Comment:
The openboxes user may not have access to system tables like this. I think that might be why you are using onError="CONTINUE", but we should find a different way of doing this if we can.

Reply:
The changeSet needs to be marked as only executing on MySQL.

Reply:
Do we want to use a check on information schema or just allow the addNotNullConstraint to fail. It feels icky to let it fail if it is something we want to ensure since it'll allow data integrity issues on some systems, so I think the info schema path is correct fi that's what Liquibase recommends.